tastytaps/taps/api.py

```diff
@@ -0,0 +1,30 @@
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+
+from .models import Brewery, Taps
+from .serializers import BrewerySerializer, TapsListSerializer, TapsSerializer
+
+
+class BreweryListCreate(generics.ListCreateAPIView):
+    permission_classes = [IsAuthenticated]
+    queryset = Brewery.objects.all()
+    serializer_class = BrewerySerializer
+
+
+class TapsMixin(object):
+    permission_classes = [IsAuthenticated]
+    queryset = Taps.objects.all()
+
+
+class TapsCreate(TapsMixin, generics.CreateAPIView):
+    serializer_class = TapsSerializer
+
+
+class TapsDetail(TapsMixin, generics.RetrieveAPIView):
+    """Returns a specific tap"""
+    serializer_class = TapsListSerializer
+
+
+class TapsList(TapsMixin, generics.ListAPIView):
+    """Returns a list of all of the taps"""
+    serializer_class = TapsListSerializer
```

Comment:
Yeah, DRF has an answer for this. I think it's `SimpleRouter` which essentially builds all the URL patterns for you based on the view:
http://www.django-rest-framework.org/api-guide/routers/#usage

You should be able to customize the list view by overriding the class' method and put your serialize in there. But, IMO, APIs should emit the same data that they accept. (This is where tests would be really helpful to see how the API works).